docs/book/user-guide/llmops-guide/evaluation/retrieval.md

```diff
@@ -0,0 +1,250 @@
+---
+description: See how the retrieval component responds to changes in the pipeline.
+---
+
+# Retrieval evaluation
+
+The retrieval component of our RAG pipeline is responsible for finding relevant
+documents or document chunks to feed into the generation component. In this
+section we'll explore how to evaluate the performance of the retrieval component
+of your RAG pipeline. We're checking how accurate the semantic search is, or in
+other words how relevant the retrieved documents are to the query.
+
+Our retrieval component takes the incoming query and converts it into a
+vector or embedded representation that can be used to search for relevant
+documents. We then use this representation to search through a corpus of
+documents and retrieve the most relevant ones.
+
+## Manual evaluation using handcrafted queries
+
+The most naive and simple way to check this would be to handcraft some queries
+where we know the specific documents needed to answer it. We can then check if
+the retrieval component is able to retrieve these documents. This is a manual
+evaluation process and can be time-consuming, but it's a good way to get a sense
+of how well the retrieval component is working. It can also be useful to target
+known edge cases or difficult queries to see how the retrieval component handles
+those known scenarios.
+
+Implementing this is pretty simple - you just need to create some queries and
+check the retrieved documents. Having tested the basic inference of our RAG
+setup quite a bit, there were some clear areas where the retrieval component
+could be improved. I looked in our documentation to find some examples where the
+information could only be found in a single page and then wrote some queries
+that would require the retrieval component to find that page. For example, the
+query "How do I get going with the Label Studio integration? What are the first
+steps?" would require the retrieval component to find [the Label Studio
+integration
+page](https://docs.zenml.io/stacks-and-components/component-guide/annotators/label-studio).
+You can see other examples in the project code base
+[here](https://github.com/zenml-io/zenml-projects/blob/e59ada173c65f79d17025aa48b87fe593229bfe8/llm-complete-guide/steps/eval_retrieval.py#L33-L34).
```

Comment:
can we make this into a small table with columns just to make it super easy to read